python/pyspark/sql/group.py

```diff
@@ -89,8 +89,15 @@ def agg(self, *exprs):
         else:
             # Columns
             assert all(isinstance(c, Column) for c in exprs), "all exprs should be Column"
-            jdf = self._jgd.agg(exprs[0]._jc,
-                                _to_seq(self.sql_ctx._sc, [c._jc for c in exprs[1:]]))
+            if isinstance(exprs[0], UDFColumn):
+                assert all(isinstance(c, UDFColumn) for c in exprs)
```

Comment:
A informative error message should be better.

Reply:
Like `all exprs should be UDFColumn"`.

Reply:
So I'm a little worried about this change, if other folks have wrapped Java UDAFs (which is reasonable since there aren't other ways to make UDAFs in PySpark before this), this seems like they won't be able to mix them. I'd suggest maybe doing what @viirya suggested bellow but instead of a failure just a warning until Spark 3.

What do y'all think?

Reply:
I am still trying to figure out the best way to dispatch this, but either way I think we won't be able to fix Java UDAF with pandas UDF.

@holdenk I am not sure what kind of warning message do you have in mind. Can you please explain?

Reply:
Ah so what your saying is you don't support mixed Python & Java UDAFs? That's certainly something which needs to be communicated in both the documentation and the error message.

Is there a reason why we don't support this?

Reply:
Answered in https://github.com/apache/spark/pull/19872#issuecomment-350139367